rollup-interface/src/state_machine/zk/mod.rs

```diff
@@ -89,7 +101,15 @@ pub trait ZkvmGuest: Zkvm {
 
 /// This trait is implemented on the struct/enum which expresses the validity condition
 pub trait ValidityCondition:
-    Serialize + DeserializeOwned + BorshDeserialize + BorshSerialize + Debug + Clone + Copy + PartialEq
+    Serialize
+    + DeserializeOwned
+    + BorshDeserialize
+    + BorshSerialize
+    + Debug
+    + Clone
+    + Copy
```

Comment:
As a lot of work already done, could we make them can you please remove `Serialize + DeserializeOwned`, because it will make harder to integrate chain state module ?

Reply:
This does need to implement `Serialize + DeserializeOwned` though - otherwise we can't pass it from host to guest. I think maybe the issue was when we tried to use `Cond` directly instead of as an associated type? So maybe this will still work?

Reply:
Yeah, let's test it when I start integrating!